src/craftingSteps.ts

```diff
@@ -99,3 +99,11 @@ function craftingStepsInner(
   treeComponents.map((component) => craftingStepsInner(component, steps, index + 1))
   return steps
 }
+
+function isMerchantWithOnlyCurrencies(step: CraftingStep): boolean {
+  return (
+    step.disciplines.length === 1 &&
+    step.disciplines[0] === 'Merchant' &&
+    step.components.every(({ type }) => type === 'Currency')
```

Comment:
Instead of checking that all components for a merchant are currencies, would checking that all components don't have any components themselves make more sense (and also protect against the case were gw2e might suggest crafting currencies (in the future))?

Reply:
That's a good idea - at generation of the crafting steps (`craftingStepsInner`) it could look at `treeComponents[].components` and sent a "hasDeepComponents" flag (or w/e) on the step. Then it could filter on that.

I was thinking it could even look at `treeComponents[].craft` to determine if the components of the merchant step are crafted themselves. Complication with that is that they might be crafted at some step and not at others (e.g. when using own materials), and we'd only want to move merchant steps to the front where none of them are. So that requires logic updates for both if/if not it finds a `stepIndex`.